python/python/bystro/covariance/_base_covariance.py

```diff
@@ -270,6 +268,45 @@ def _conditional_score(covariance, X, idxs, weights=None):
     return avg_score
 
 
+def _conditional_score_sherman_woodbury(Lambda, W, X, idxs, weights=None):
+    """
+    Returns the predictive log-likelihood of a subset of data.
+
+    mean(log p(X[idx==1]|X[idx==0],covariance))
+
+    Parameters
+    ----------
+    Lambda : np.array-like,(p,p)
```

Comment:
Since you're already adding the type in parameters, better to add types as well; e.g. 

```py
from numpy.typing import NDArray

if _conditional_score_sherman_woodbury(Lambda: NDArray, W: NDarray, X: NDArray, idxs: NDArray, weights: NDArray | None = None):
```

I also don't think you actually support ArrayLike, because you rely on NDArray semantics directly in `_conditional_score_samples_sherman_woodbury`. To support ArrayLike you would need to cast each argument to  NDArray first, e.g. `np.array(Lambda)`.


With Github Copilot and VS Code, the type would probably be filled in for you